File: IntelligentSampleEnginePOC.API/IntelligentSampleEnginePOC.API.Core/Model/TargetAudience.cs
using System.Text.Json.Serialization;

namespace IntelligentSampleEnginePOC.API.Core.Model
{
    public class TargetAudience
    {
        [JsonPropertyName("id")]
        public long Id { get; set; }
        
        [JsonPropertyName("name")]
        public string Name { get; set; }
        
        [JsonPropertyName("audienceNumber")]
        public int AudienceNumber { get; set; }
        
        [JsonPropertyName("estimatedIR")]
        public int EstimatedIR { get; set; }
        
        [JsonPropertyName("estimatedLOI")]
        public int EstimatedLOI { get; set; }
        
        [JsonPropertyName("limit")]
        public int Limit { get; set; }
        
        [JsonPropertyName("limitType")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public int? LimitType { get; set; }
        
        [JsonPropertyName("testingUrl")]
        // [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public string? TestingUrl { get; set; }
        
        [JsonPropertyName("liveUrl")]
        // [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public string? LiveUrl { get; set; }

        [JsonPropertyName("qualifications")]
        public List<Qualification> Qualifications { get; set; } = new();

        [JsonPropertyName("quotas")]
        public List<Quota> Quotas { get; set; } = new();
    }
}

File: ise-ui-vue/src/models/targetAudience.ts
import { z } from "zod";

export const TargetAudienceSchema = z
	.object({
		id: z.number().int().positive(),
		name: z.string().min(1),
		audienceNumber: z.number().int().positive(),
		estimatedIR: z.number().int().nonnegative(),
		estimatedLOI: z.number().int().positive(),
		limit: z.number().int().nonnegative(),
		limitType: z.number().int().positive().nullable().optional(),
		testingUrl: z.string().min(1).optional().nullable(),
		liveUrl: z.string().min(1).optional().nullable(),
		qualifications: z.object({}).array().nullable(), // Placeholder
		quotas: z.object({}).array().nullable(), // Placeholder
	})
	.strict();

export type TargetAudience = z.infer<typeof TargetAudienceSchema>;
